pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -96,8 +96,31 @@ private CalciteSqlParser() {
   private static final Pattern OPTIONS_REGEX_PATTEN =
       Pattern.compile("option\\s*\\(([^\\)]+)\\)", Pattern.CASE_INSENSITIVE);
 
+  /**
+   * Checks for the presence of semicolon in the sql query and modifies the query accordingly
+   *
+   * @param sql sql query
+   * @return sql query without semicolons
+   *
+   */
+  private static String removeTerminatingSemicolon(String sql) {
+    // Check if the query has semicolon
+    int semiColonIndex = sql.lastIndexOf(';');
+    int sqlLength = sql.length();
+    boolean stripSemiColon = semiColonIndex >= 0;
+
+    // check if semicolons are followed by whitespaces, only then termination should be done
+    for (int i = semiColonIndex + 1; i < sqlLength && stripSemiColon; i++) {
```

Comment:
Let's trim the sql first before checking for the last semicolon. This can help remove the unnecessary whitespace from the query, and we can simply check the character at the last index after trimming

Reply:
Updated the PR as per the review comment